haversine-processing-engine/src/consumer.ts

```diff
@@ -0,0 +1,48 @@
+import * as kafka from 'kafka-node';
+import { log, haversine } from './utils';
+
+import { kafkaType } from "./constants";
+
+const createKafkaClient = (kafkaConfig: kafkaType) => {
+  try {
+    const client = new kafka.KafkaClient({
+      kafkaHost: kafkaConfig.kafkaEndpoint
+    });
+    const consumer = createKafkaConsumer(kafkaConfig, client)
+    consumer.on('message',
+      async (message: kafka.Message) => handleMessage(message, kafkaConfig.topic))
+      .commit((error: any, data: any) => {
+        if (error) {
+          log(`Error while committing message: ${error}, ${kafkaConfig.topic}`);
+        }
+      });
+  } catch (e) {
+    log(`Unhandled exception with details: ${e}, ${kafkaConfig.topic}`);
+  }
+}
+
+const handleMessage = async (message: kafka.Message, topic: string) => {
+  const parsedData = JSON.parse(message.value.toString());
+  console.log(`message received: ${message.value.toString()}`);
```

Comment:
probably want to remove console logs for Prod... also, when timestamping to figure out how long your process is running, these guys are very heavy.